lib/services/watcher_test.go

```diff
@@ -1042,3 +1042,119 @@ func newAccessRequest(t *testing.T, name string) types.AccessRequest {
 	require.NoError(t, err)
 	return accessRequest
 }
+
+// TestSAMLIdPServiceProviderWatcher tests that SAML IdP service provider resource watcher properly receives
+// and dispatches updates to SAML IdP service provider resources.
+func TestSAMLIdPServiceProviderWatcher(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	clock := clockwork.NewFakeClock()
+
+	bk, err := memory.New(memory.Config{
+		Context: ctx,
+		Clock:   clock,
+	})
+	require.NoError(t, err)
+
+	type client struct {
+		services.SAMLIdPServiceProviders
+		types.Events
```

Comment:
nit: Name the fields.

Reply:
This is less doable -- I'm basically using this as a surrogate for an `auth.Client`, so I'll keep this as composition.